### ISSUE TITLE
bug: Ionic Angular Router Link Prop Reload the Whole Page During Navigation

### ISSUE THREAD

**Issue by yoelpc4** [ionic-team/ionic-framework#25076]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

When the component ion-card, ion-item, or ion-button has a `routerLink` prop is clicked, it gonna navigates to the desired page but also reloads the whole page unlike when we do navigation via `this.router.navigateByUrl`, this means we lost the application state for every navigation.

### Expected Behavior

When the component ion-card, ion-item, or ion-button has a `routerLink` prop is clicked, it is supposed to navigate to the desired page but without reloading the whole page, so that the application state is still the same as when we do navigation via `this.router.navigateByUrl`.

### Steps to Reproduce

- Clone the project from provided code reproduction URL below, then serve the project on the web.
- Click the ion-card, ion-item, and ion-button with `Go to Dashboard via Router Link` text to see the current behavior.
- Click the ion-card, ion-item, and ion-button with `Go to Dashboard via Navigate by Url` text to see the expected behavior.

### Code Reproduction URL

https://github.com/yoelpc4/router-link-issue

### Ionic Info

```
Ionic:

   Ionic CLI                     : 6.19.0 (/Users/yoelpc4/.nvm/versions/node/v16.14.2/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.15
   @angular-devkit/build-angular : 13.2.6
   @angular-devkit/schematics    : 13.2.6
   @angular/cli                  : 13.2.6
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.4.3
   @capacitor/android : not installed
   @capacitor/core    : 3.4.3
   @capacitor/ios     : not installed

Utility:

   cordova-res : 0.15.4
   native-run  : 1.5.0

System:

   NodeJS : v16.14.2 (/Users/yoelpc4/.nvm/versions/node/v16.14.2/bin/node)
   npm    : 8.5.0
   OS     : macOS Monterey
```

### Additional Information

The reproduction is using fresh installed ionic angular via the `ionic start` command without any alteration to the package's version on package.json.

I also have a current project with this ionic info as follows:

```
Ionic:

   Ionic CLI                     : 6.19.0 (/Users/yoelpc4/.nvm/versions/node/v16.14.2/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.15
   @angular-devkit/build-angular : 13.3.2
   @angular-devkit/schematics    : 13.3.2
   @angular/cli                  : 13.3.2
   @ionic/angular-toolkit        : 6.1.0

Capacitor:

   Capacitor CLI      : 3.4.3
   @capacitor/android : 3.4.3
   @capacitor/core    : 3.4.3
   @capacitor/ios     : 3.4.3

Utility:

   cordova-res : 0.15.4
   native-run  : 1.5.0

System:

   NodeJS : v16.14.2 (/Users/yoelpc4/.nvm/versions/node/v16.14.2/bin/node)
   npm    : 8.5.0
   OS     : macOS Monterey
```

 I'm also gonna test the bug fix later with my current project in case of different version still has this issue.

**Comment by liamdebeasi** [comment]
Thanks for the issue. We fixed this earlier today in https://github.com/ionic-team/ionic-framework/commit/1c26e9b9b0fc45a8691e972fe17a168f89a27a79.

I am going to close this, but there is a dev build in https://github.com/ionic-team/ionic-framework/pull/25014#issuecomment-1091743772 that you can use to get unblocked with development. We are planning to ship this fix by next week at the latest.

**Comment by danielehrhardt** [comment]
> Thanks for the issue. We fixed this earlier today in https://github.com/ionic-team/ionic-framework/commit/1c26e9b9b0fc45a8691e972fe17a168f89a27a79.
> 
> I am going to close this, but there is a dev build in https://github.com/ionic-team/ionic-framework/pull/25014#issuecomment-1091743772 that you can use to get unblocked with development. We are planning to ship this fix by next week at the latest.

I my opinion it should be handled as hotfix. So please create a release for this. It took me multiple hours to find the root cause. 
^ in package.json

**Comment by liamdebeasi** [comment]
There will be a hot fix. If this is impacting an app in production, you can safely roll back to 6.0.14 for now.

We have a team policy to wait at least 24 hours after a fix has been created before releasing. This allows for the community to test and provide feedback on the dev build we have created. (It would be quite bad if the fix for the bug introduced another bug 😄 ).

**Comment by liamdebeasi** [comment]
This issue is resolved in Ionic 6.0.16. Please update your apps to receive a fix:

```
npm install @ionic/angular@6.0.16
```

**Comment by yoelpc4** [comment]
OK I can confirm the [hotfix release](https://github.com/ionic-team/ionic-framework/releases/tag/v6.0.16) is working as intended @liamdebeasi @danielehrhardt
